src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -680,6 +680,38 @@ struct info
   }
 };
 
+struct flash_address
+{
+  using result_type = uint64_t;
+
+  static result_type
+  user(const xrt_core::device* device, key_type)
+  {
+	  return 0;
+  }
+
+  static result_type
+  mgmt(const xrt_core::device* device, key_type)
+  {
+    auto init_addr = [](const xrt_core::device* dev) {
+      uint64_t addr;
+      mgmtpf::get_flash_addr(dev->get_mgmt_handle(), addr);
+      return addr;
+    };
+
+    static std::map<const xrt_core::device*, uint64_t> info_map;
```

Comment:
Please add some document regarding the purpose behind the map and mutex (e.g., caching and the need for thread safety).